mmdet/core/bbox/assigners/center_region_assigner.py

```diff
@@ -0,0 +1,297 @@
+                map indices and the second column the shadowed gt indices
+        """
+        num_bboxes, num_gts = is_bbox_in_gt_core.shape
+
+        if gt_priority is None:
+            gt_priority = torch.arange(num_gts).to(is_bbox_in_gt_core.device)
+            # the bigger, the more preferable to be assigned
+        # the assigned inds are by default 0 (background)
+        assigned_gt_inds = is_bbox_in_gt_core.new_full((num_bboxes, ),
+                                                       0,
+                                                       dtype=torch.long)
+        inds_of_match = torch.any(is_bbox_in_gt_core, dim=1)
+        # matched  bboxes (to any gt)
+        inds_of_shadow = torch.any(is_bbox_in_gt_shadow, dim=1)
+        # Ignored indices are assigned to be background. But the corresponding
+        #   label is shadowed during loss calculation, which is done through
+        #   shadowed_gt_inds
+        assigned_gt_inds[inds_of_shadow] = 0
+        shadowed_gt_inds = torch.nonzero(is_bbox_in_gt_shadow)
+        if is_bbox_in_gt_core.sum() == 0:  # No gt match
+            shadowed_gt_inds[:, 1] += 1  # 1-based. For consistency issue
+            return assigned_gt_inds, shadowed_gt_inds
+
+        # The priority of each prior box and gt pair. If one prior box is
+        #  matched bo multiple gts. Only the pair with the highest priority
+        #  is saved
+        pair_priority = is_bbox_in_gt_core.new_full((num_bboxes, num_gts),
+                                                    -1,
+                                                    dtype=torch.long)
+
+        # Each bbox could match with multiple gts.
+        # The following codes deal with this situation
+
+        # Whether a bbox match a gt,  bool tensor, shape [num_match, num_gt]
+        matched_bbox_and_gt_correspondence = is_bbox_in_gt_core[inds_of_match]
+        # The matched gt index of each positive bbox. Length >= num_match,
+        #  since one bbox could match multiple gts
+        matched_bbox_gt_inds =\
+            torch.nonzero(matched_bbox_and_gt_correspondence)[:, 1]
+        # Assign priority to each bbox-gt pair.
+        pair_priority[is_bbox_in_gt_core] = gt_priority[matched_bbox_gt_inds]
+        _, argmax_priority = pair_priority[inds_of_match].max(dim=1)
+        # the maximum shape [num_match]
+        # effective indices.
+        assigned_gt_inds[inds_of_match] = argmax_priority + 1  # 1-based
+        # Zero-out the assigned pixels to filter the shadowed gt indices
+        is_bbox_in_gt_core[inds_of_match, argmax_priority] = 0
+        # Concat the shadowed indices due to overlapping with that out side of
+        #   effective scale. shape: [total_num_ignore, 2]
+        shadowed_gt_inds = torch.cat(
+            (shadowed_gt_inds, torch.nonzero(is_bbox_in_gt_core)), dim=0)
+        is_bbox_in_gt_core[inds_of_match, argmax_priority] = 1  # no change
+        shadowed_gt_inds[:, 1] += 1  # 1-based. For consistency issue
```

Comment:
We'd better use 0-based class cat in our code base.

Reply:
Oh, this is not class category. It is the gt index (from 1 to num_gts), which is 1-based and should be consistent with assigned_gt_inds